### PR TITLE
Docker

### PR DESCRIPTION
**Pull request by mkondapalli**
#62 ..Checked in changes related to build Statspoller Docker image.

Two ways to build the image
1) MVN Plugin 
    - Build : mvn clean package docker:build  
    - Push : mvn docker:push
2) Docker Compose
    - docker-compose up --build -d statsp

### CODE REVIEW

**Comment by mkondapalli** [comment]
@johnt337 ..This needs to be reviewed.

File: .gitignore [hunk 1]
```diff
@@ -8,4 +8,4 @@
 /logs/*
 !/logs/ignore.me
 /output/*
-!/output/ignore.me
+!/output/ignore.me
```
Comment (dhoer) [inline .gitignore]: Please add a blank line at the end of this file

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+COPY conf /statsp/conf
+
+RUN chown -R statsp:statsp /statsp
+
+USER statsp
+
+HEALTHCHECK --interval=5m --timeout=3s \
+  CMD ps -ef | grep StatsPoller.jar || exit 1
+
+CMD ["java","-jar", "StatsPoller.jar"]
```
Comment (dhoer) [inline Dockerfile]: The java piece could be the entrypoint e.g., `ENTRYPOINT ["JAVA"]`

File: docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+version: "2"
+services:
+  statsp:
+    build:
+      context: .
+    depends_on:
+    - httpd
+    volumes:
+    - ./conf:/statsd/conf
```
Comment (dhoer) [inline docker-compose.yml]: This line could be dropped since line 13 of Dockerfile already copies the files.

File: docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,26 @@
+FROM openjdk:8-jre
```
Comment (dhoer) [inline docker/Dockerfile]: If this file serves a different purpose than the main one, please state that purpose in a comment at the top.

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,22 @@
+FROM openjdk:8-jre
```
Comment (dhoer) [inline Dockerfile]: Do you want to work together and get this working with alpine?

**Comment by mkondapalli** [comment]
Sure Dennis. May be this afternoon ?

File: Dockerfile (hunk 2 above)
Comment (johnt337) [inline Dockerfile]: You might want to `grep -v grep` as it may get a result from the grep its self.

File: docker-compose.yml (hunk 3 above)
Comment (johnt337) [inline docker-compose.yml]: As @dhoer commented, did you mean to override the bundled version with a local version at runtime?

File: docker/Dockerfile [hunk 6]
```diff
@@ -1,12 +1,19 @@
...
 
-USER root
+#USER root
 
 WORKDIR /statsp
 
 COPY . /statsp
 
-RUN groupadd -r statsp && useradd -r -g statsp statsp
+RUN addgroup statsp && adduser -D -G statsp statsp
```
Comment (johnt337) [inline docker/Dockerfile]: You should specify `USER statps` after this if you intend to run the entrypoint as this user.

File: docker/Dockerfile [hunk 7]
```diff
@@ -1,12 +1,19 @@
...
 
 RUN mkdir /statsp/output /statsp/logs
```
Comment (johnt337) [inline docker/Dockerfile]: You should specify USER statsp after this if you intend to run the entrypoint as this user.

**Pull request merged**